discovery-provider/src/utils/csv_writer.py

```diff
@@ -0,0 +1,25 @@
+import csv
+from io import StringIO
+from typing import List
+
+
+# Write CSV to string buffer before returning the contents of the string.
+# This function takes a list of dictionaries, where each dictionary represents a row.
+# Example:
+#   rows = [{"name": "John", "age": 30}, {"name": "Jane", "age": 25}]
+#   write_csv(rows)
+# Returns:
+#    name,age
+#    John,30
+#    Jane,25
+def write_csv(rows: List[dict]):
```

Comment:
super duper nitpick feel free to ignore: "write" makes me think there's a file somewhere, maybe `write_csv_string` or something?